### PR TITLE
Additional bandit analysis tests

### PR DESCRIPTION
**Pull request by dctaf**
These tests improve coverage of corner cases in bandit analysis and will serve to ensure that upcoming performance-related changes do not alter user-visible behavior of the API.

### CODE REVIEW

File: src/xngin/apiserver/snapshots/test_snapshotter.py [hunk 1]
```diff
@@ -599,21 +666,112 @@ async def test_create_snapshot_cmab_with_zero_draws_succeeds(
...
+        assert assignment_response.status == 200, assignment_response.data
+        if outcome is not None:
+            eclient.update_bandit_arm_with_participant_outcome(
+                api_key=testing_datasource.key,
+                body=UpdateBanditArmOutcomeRequest(outcome=outcome),
+                experiment_id=experiment_id,
+                participant_id=participant_id,
+            )
+
+    expected_contexts = [1.0 if context.value_type == ContextType.BINARY else 7.0 / 3.0 for context in sorted_contexts]
```
Comment (markbotterill) [inline src/xngin/apiserver/snapshots/test_snapshotter.py]: Nit: Can we we add a one-liner comment just explaining the logic here? e.g. # BINARY context uses rounded mean of assignments; REAL_VALUED uses mean

Comment (markbotterill) [reply src/xngin/apiserver/snapshots/test_snapshotter.py]: Just makes it a bit easier to realize where the 7.0 comes from. Additionally, we could write that 7.0 as (1 + 4 + 2)

Comment (markbotterill) [reply src/xngin/apiserver/snapshots/test_snapshotter.py]: Same for the binary: BINARY (1.0, 0.0, 1.0) → mean = 0.667 → ceil of (-0.5 - 0.667)=0.167 -> 1. Just makes scanning a tiny bit easier!

**Review by markbotterill — APPROVED** [review]
Looks great! Just added a tiny comment on test setup readability :)

File: src/xngin/apiserver/snapshots/test_snapshotter.py (hunk 1 above)
Comment (dctaf) [reply src/xngin/apiserver/snapshots/test_snapshotter.py]: Done. Thank you!
